src/mongocrypt-key-broker.c

```diff
@@ -817,6 +817,10 @@ mongocrypt_kms_ctx_t *_mongocrypt_key_broker_next_kms(_mongocrypt_key_broker_t *
         return NULL;
     }
 
+    if (kb->keys_returned->kms.should_retry) {
```

Comment:
`keys_returned` is a linked list. I expect this only retries the first KMS request if there are multiple, and may need to loop:
```c
// Check if any requests need retry
for (key_returned_t *ptr = kb->keys_returned; ptr != NULL; ptr = ptr->next) {
    if (ptr->kms.should_retry) {
        ptr->kms.should_retry = false;
        return &ptr->kms;
    }
}
```

Here is a suggested test of multiple KMS requests: https://github.com/mongodb/libmongocrypt/commit/fbabc573219915d0997b44127bc23693773317ab#diff-70d771e7c35927baab41e7a4472474753d07fbb2cccf73640626663ad3a3b771R5095. Test data was generated with the `csfle` utility.